resources/todomvc/architecture-examples/svelte/src/Item.svelte

```diff
@@ -0,0 +1,55 @@
+<script>
+    import { createEventDispatcher, tick } from 'svelte';
+
+    let editing = false;
+
+    const dispatch = createEventDispatcher();
+
+    function removeItem() {
+        dispatch('removeItem');
+    }
+
+    function startEdit() {
+       editing = true;
+    }
+
+    function handleEdit(event) {
+        if (event.key === "Enter")
+            event.target.blur();
+        else if (event.key === "Escape") 
+            editing = false;
+    }
+
+    function updateItem(event) {
+        if (!editing) return;
+        dispatch('updateItem', { text: event.target.value });
+        editing = false;
+    }
+
+    function toggleItem(event) {
+        dispatch('toggleItem', { checked: event.target.checked });
+    }
```

Comment:
I believe that the toggle operation could be done directly here.

Reply:
@julienw - do you mean an inline function like this in the item component?
`<input class="toggle" type="checkbox" on:change={(e) => item.completed = e.target.checked } checked={item.completed} />`

I tried that and it didn't work, maybe since I don't re-assign the item data in the parent?

Reply:
yeah, seems like we'd need to force an update in the parent and that would make the toggle update mood...
thanks for the above 😄